src/App.jsx

```diff
@@ -14,57 +15,67 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
+  const buttonClear = () => {
+    setSelectedGood('');
+  };
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood && `${selectedGood} is selected`}
+        {selectedGood ? (
+          <button
+            onClick={buttonClear}
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+          />
+        ) : (
+          'No goods selected'
+        )}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
+      <table className="table">
+        <tbody>
+          {goods.map(word => (
+            <tr
+              data-cy="Good"
+              key={word}
+              className={
+                selectedGood === word && 'has-background-success-light'
+              }
             >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+              <td>
+                {selectedGood !== word ? (
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className={`button ${word === selectedGood && 'has-background-success-light'}`}
+                    onClick={() => setSelectedGood(word)}
+                  >
+                    +
+                  </button>
+                ) : (
+                  <button
+                    data-cy="RemoveButton"
+                    type="button"
+                    className="button is-info"
+                    onClick={buttonClear}
+                  >
+                    -
+                  </button>
+                )}
```

Comment:
It is better to alter the properties of the component than to alter the whole component
Actually, you have started this

```suggestion
 <button
    type="button"
    data-cy="{word === selectedGood ? 'RemoveButton' : 'AddButton'}
    className = {...}
    onClick={() => setSelectedGood(word === selectedGood ? '' : 'word)}
         >
        {...}
   </button>
          
```

Reply:
Thanks for this idea!
 It looks good, but I tried to do it, and I got confused about it, and I decided to put my solution made using the KISS method because it is more understandable for me. 
I think later when I understand this topic better, I will do it in the way you showed